### DISCUSSION TITLE
Basic ray-traced reflections

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#14952] [Show and tell]
@DougM

![bb_Sphere](https://github.com/o3de/o3de/assets/5120856/3fe4678f-887f-4f21-90e4-ddafcf4c66b9)
![AMeng](https://user-images.githubusercontent.com/5120856/225647610-4b9ab5db-5721-4c16-aff8-e6042e63a6bc.jpg)
![MetalRtReflections](https://user-images.githubusercontent.com/5120856/223337600-e34eac9f-6d9d-4d9c-bcbf-91c9a18bcd83.jpg)

Features
- Specular RT reflections, replacing SSR
- Reflection rays from G-buffer
- **Ray-traced shadows in reflections**
- **Recursive** reflections
- IBL
- RayTracingPass derived class
- **2 miss shaders**

**I'm not able to make it work for Vulkan, seems SRG has some issue.** But material preview is more correct in Vulkan.
- Solution: #define USE_BINDLESS_SRG 0

**Comment by AMZN-alexpete** [comment]
Look great!

**Comment by ohmaya** [comment]
Using full-fledged **screen-space diffuse shading (all lights, GI, IBL & soft shadows)** when possible, blending between the 2 based on screen edges, normals, and depths:
![SSDiffuse](https://user-images.githubusercontent.com/5120856/229291479-c2f8a645-f379-4b7d-9d24-96af27b5dc50.jpg)
![Box](https://user-images.githubusercontent.com/5120856/231978558-4265a718-e39b-4ba0-b0f4-7fd2975639d8.jpg)

**Comment by ohmaya** [comment]
<img width="838" alt="ChinaStyle04" src="https://github.com/o3de/o3de/assets/5120856/c8bcc911-7cba-41ec-8d18-e49d21a00e50">

- Cutout
- Double-sided

**Comment by aldenpang** [comment]
what a nice work!

**Comment by ohmaya** [comment]
![TransparentReflections](https://github.com/o3de/o3de/assets/5120856/d500aad6-691a-42a2-bbca-1c1d3f6cce4b)
![TransparentReflections02](https://github.com/o3de/o3de/assets/5120856/14cb8c88-a0cd-4dd9-999a-c3af7038fc00)
**Transparent reflections**
- Single-layer reflections (Control/Northlight engine style)
- Identifying transparent objects by instance masks
- Utilizing Opaque DiffuseSpecularMerge and saving tracing where possible
